src/TestFramework/Coverage/CodeCoverageData.php

```diff
@@ -98,73 +97,32 @@ static function ($testMethods) {
         return \count($coveredLineTestMethods) > 0;
     }
 
-    public function hasTestsOnLine(string $filePath, int $line): bool
-    {
-        $coverageData = $this->getCoverage();
-
-        if (!isset($coverageData[$filePath])) {
-            return false;
-        }
-
-        if (!isset($coverageData[$filePath]->byLine[$line])) {
-            return false;
-        }
-
-        return !empty($coverageData[$filePath]->byLine[$line]);
-    }
-
-    public function hasExecutedMethodOnLine(string $filePath, int $line): bool
```

Comment:
1. Why don't we need it anymore? 
2. where are values `executed`, `coverage` used from `CoverageMethodData` now? If they are not needed, we shouldn't save them in that DTO class

Reply:
> Why don't we need it anymore?

This function determines if the method is tested or not, if it is tested, we do (almost) the same thing again to determine what test methods test it. Otherwise we skip that.

The method that determines the tests has almost duplicate logic, so instead of checking it first, we simply let that method resolve it.

Where if the method has no tests, it simply returns an empty array (or generator).